### PR TITLE
firefox_desktop_to_glam

### PR DESCRIPTION
**Pull request by alekhyamoz**
Work In Progress, Draft PR for FOG into GLAM
[Reference doc](https://docs.google.com/document/d/1jMyXi9fdgRgQVavi1VWD8mD1svBeSWuw8frt3--yJsY/edit#)

### CODE REVIEW

File: bigquery_etl/glam/templates/logical_app_id/firefox_desktop_glam_beta__view_clients_daily_histogram_aggregates_v1.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  SELECT
+    *
+  FROM
+    `{{ project }}`.glam_etl.firefox_desktop__view_clients_daily_histogram_aggregates_v1
+  WHERE
+    channel = 'beta'
+)
+SELECT
+  * EXCEPT (app_build_id, channel),
+  `alekhya-test-1-322715.udf.build_seconds_to_hour`(app_build_id) AS app_build_id,
```
Comment (emtwo) [inline bigquery_etl/glam/templates/logical_app_id/firefox_desktop_glam_beta__view_clients_daily_histogram_aggregates_v1.sql:14]: We'll need to add this udf to `shared-prod`

File: sql/glam-fenix-dev/glam_etl/firefox_desktop__view_clients_daily_scalar_aggregates_v1/view.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- view for firefox_desktop__view_clients_daily_scalar_aggregates_v1;
+-- View to union daily scalar aggregates with date partitioning
+CREATE OR REPLACE VIEW
+  `glam-fenix-dev.glam_etl.firefox_desktop__view_clients_daily_scalar_aggregates_v1`
```
Comment (emtwo) [inline sql/glam-fenix-dev/glam_etl/firefox_desktop__view_clients_daily_scalar_aggregates_v1/view.sql:4]: `glam-fenix-dev` doesn't apply here for desktop. I think the folder that the sql is generated into needs to be changed as well.

File: script/glam/test/test_glean_firefox_desktop_glam_nightly.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+set -e
+
+project=${PROJECT:-glam-fenix-dev}
+skip_generate=${SKIP_GENERATE:-false}
+skip_daily=${SKIP_DAILY:-false}
+generate_only=${GENERATE_ONLY:-false}
+# NOTE: there are three app_ids that we must look at for historical context. For
+# the purpose of this script, it is sufficient to look only at what is currently
+# "fenix nightly". We must have at least one table scalar/histogram tables for
```
Comment (emtwo) [inline script/glam/test/test_glean_firefox_desktop_glam_nightly.sh:12]: desktop

File: sql/glam-fenix-dev/glam_etl/firefox_desktop_glam_nightly__clients_scalar_aggregates_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  FROM
+    glam_etl.firefox_desktop_glam_nightly__clients_scalar_aggregates_v1 AS scalar_aggs
+  LEFT JOIN
+    glam_etl.firefox_desktop_glam_nightly__latest_versions_v1
+  USING
+    (channel)
+  WHERE
+      -- allow for builds to be slighly ahead of the current submission date, to
+      -- account for a reasonable amount of clock skew
+    mozfun.glam.build_hour_to_datetime(app_build_id) < DATE_ADD(@submission_date, INTERVAL 3 day)
```
Comment (emtwo) [inline sql/glam-fenix-dev/glam_etl/firefox_desktop_glam_nightly__clients_scalar_aggregates_v1/query.sql:224]: Does this work? I think you said the `app_build_id` is in seconds for FOG?

File: bigquery_etl/glam/templates/logical_app_id/firefox_desktop_glam_beta__view_clients_daily_histogram_aggregates_v1.sql (hunk 1 above)
Comment (alekhyamoz) [reply bigquery_etl/glam/templates/logical_app_id/firefox_desktop_glam_beta__view_clients_daily_histogram_aggregates_v1.sql:14]: yes that was my long term goal.alekhya-test-1-322715  was just to generate the etl scripts

File: sql/glam-fenix-dev/glam_etl/firefox_desktop_glam_nightly__clients_scalar_aggregates_v1/query.sql (hunk 4 above)
Comment (alekhyamoz) [reply sql/glam-fenix-dev/glam_etl/firefox_desktop_glam_nightly__clients_scalar_aggregates_v1/query.sql:224]: Yes by the time it reaches to this function the app_build_id is already truncated to hour (yyyymmddhh)

File: sql/glam-fenix-dev/glam_etl/firefox_desktop__view_clients_daily_scalar_aggregates_v1/view.sql (hunk 2 above)
Comment (alekhyamoz) [reply sql/glam-fenix-dev/glam_etl/firefox_desktop__view_clients_daily_scalar_aggregates_v1/view.sql:4]: good point. I will work on this

File: script/glam/test/test_glean_firefox_desktop_glam_nightly.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+generate_only=${GENERATE_ONLY:-false}
+# NOTE: there are three app_ids that we must look at for historical context. For
+# the purpose of this script, it is sufficient to look only at what is currently
+# "firefox desktop". We must have at least one table scalar/histogram tables for
+# each of the referenced tables in the view. We'll keep all pings for
+# firefox_desktop, and only the metrics ping for the others.
+app_ids=(
+    "firefox_desktop"
+)
+logical_app_id="firefox_desktop_glam_nightly"
```
Comment (emtwo) [inline script/glam/test/test_glean_firefox_desktop_glam_nightly.sh:18]: I think this file is identical to the fenix one except for these two lines above right? Can we use the same test file for both and just have an if statement here depending and set the values of `app_ids` and `local_app_id` base on an additional input param that specifies desktop or fenix?

**Review by emtwo — APPROVED** [review]
looks good to me pending the one comment about unifying the tests. It would make changes to the tests in the future easier to maintain if its all in one place.

**Pull request merged**